### PR TITLE
Type of financial institution (Phase 1.5)

### LINKED ISSUE #455: Type of Financial Institution - get it in the workflow!
Doesn't have to be pretty, just get it in.

Phase 1
- [ ] Don't auto create Filing, they have to click "Start"
- [ ] If they don't have FI Types => Route to FI Types 
- [ ] If they do have FI Types => Route to Upload
- [ ] ~~Filing steps~~ Doesn't show in Filing steps
- [ ] On Submit => Route to Upload
 
Phase 2
- [ ] API integration

### PR DESCRIPTION
**Pull request by meissadia**
Closes #455 

## Changes

- Integrates Type of FI form into the workflow
- Updates Type of FI form content
- Prevents creation of a Filing for the institution until they've clicked "Start filing"

## How to test this PR

1. Create an institution without any sbl types (Example JSON below)
2. Associate LEI with the domain (local example)
   - POST `{{regtech_url}}/v1/institutions/{{lei}}/domains`
   - body:
    ```
    [
      {
        "domain": "local.host"
      }
    ]
    ```
3. Click on `Start filing` for that institution (Card heading should read "Start the filing process (institution)", which is a temporary adjustment to help identify if the user will be routed to Types of FI `(institution)` or to the Upload page `(filing)`)
4. Should be directed to `/institutions/{your-lei}/type/2024`
5. You can fill out the form (not required to `Save and continue`)
6. Click `Save and continue`
7. Should be directed to an intermediary page that creates a Filing for that institution (`Loading filing data...`)
7. Once Filing is created, should be auto redirected to the Upload page for that institution

### Flow for institution without existing SBL types
https://github.com/cfpb/sbl-frontend/assets/2592907/e8f3550e-1574-4ec1-a7f4-716b1b3f2fef

### FILING HOME: Institution that DOES NOT already have SBL types (`institution`) (will be routed to Type of FI form)
Screenshot 2024-05-09 at 10 55 17 AM

### FILING HOME: Institution that already has SBL types (`filing`) but has not yet started the filing process
Screenshot 2024-05-09 at 10 55 10 AM

### Type of FI form
screencapture-localhost-8899-institution-123456789MEISBANK808-type-2024-2024-05-09-10_59_31

## Institution without sbl types
```
{
  "name": "Meis Bank 808",
  "lei": "123456789MEISBANK808",
  "is_active": true,
  "tax_id": "88-9999991",
  "rssd_id": 9991,
  "primary_federal_regulator_id": "OCC",
  "hmda_institution_type_id": "1",
  "sbl_institution_types": [],
  "hq_address_street_1": "Test Address Street 1",
  "hq_address_street_2": "",
  "hq_address_city": "Test City 1",
  "hq_address_state_code": "GA",
  "hq_address_zip": "00000",
  "parent_lei": "PARENTTESTBANK999",
  "parent_legal_name": "PARENT TEST BANK 999",
  "parent_rssd_id": 99945,
  "top_holder_lei": "TOPHOLDERLEI999",
  "top_holder_legal_name": "TOP HOLDER LEI 999",
  "top_holder_rssd_id": 999456
}
```

### CODE REVIEW

File: src/pages/Filing/FilingApp/FilingCreate.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  if (isLoading) return <LoadingContent message='Loading filing data...' />;
+
+  /** Filing exists */
+  if (
+    filing ??
+    Number(error?.response?.status) === FILING_STATUS_CODE_FILING_EXISTS
+  ) {
+    // Note: React was complaining about setState during render. This setTimeout seems to resolve the issue.
+    setTimeout(() => navigate(`/filing/${year}/${lei}/upload`), 0);
```
Comment (billhimmelsbach) [inline src/pages/Filing/FilingApp/FilingCreate.tsx:28]: Can we make a comment/post-mvp ticket here to look into this a little more?

**Review by billhimmelsbach — COMMENTED** [review]
Heyo! Generally, this is looking good @meissadia! I'm mostly asking for tickets and TODOs, since I know this branch is the base for others. I don't think we need to fix everything here, but I'd like to revisit some of these things later. 👍

File: src/pages/TypesFinancialInstitutions/index.tsx [hunk 2]
```diff
@@ -1,47 +1,94 @@
...
+
+  if (isError)
+    return <AlertApiUnavailable message='Unable to load institution data' />;
+
+  const onSubmit = async (): void => {
+    // TODO: Form validation
+    const passesValidation = true; // await trigger();
+    if (passesValidation) {
+      // TODO: API integration
+      console.log('Submit:', getValues());
```
[the remark this answers is not in the document]
Comment (meissadia) [reply src/pages/TypesFinancialInstitutions/index.tsx:57]: Yes, cleared next PR where API integration is implemented.

File: src/utils/useCreateFiling.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+const useCreateFiling = (
+  lei: InstitutionDetailsApiType['lei'],
+  filingPeriod: FilingPeriodType,
+): UseQueryResult<FilingType> => {
+  const auth = useSblAuth();
+
+  if (!lei) throwMissingParameter('lei');
+  if (!filingPeriod) throwMissingParameter('filingPeriod');
+
+  return useQuery({
```
[the remark this answers is not in the document]
Comment (meissadia) [reply src/utils/useCreateFiling.ts:21]: Follow-up ticket: https://github.com/cfpb/sbl-frontend/issues/507

File: src/pages/Filing/FilingApp/FilingCreate.tsx (hunk 1 above)
Comment (meissadia) [reply src/pages/Filing/FilingApp/FilingCreate.tsx:28]: Follow-up ticket: https://github.com/cfpb/sbl-frontend/issues/508

**Review by billhimmelsbach — APPROVED** [review]
Thanks for the tickets! 🚀

**Pull request merged**